common/tools/dev-tool/src/commands/samples/checkNodeVersions.ts

```diff
@@ -67,20 +67,39 @@ async function cleanup(
   await deleteDockerImages(dockerImageNames);
 }
 
+function findSamplesDir(samplesDir: string, rootDir: string): string {
+  const dirs = [];
+  for (const file of fs.readdirSync(samplesDir)) {
+    const stats = fs.statSync(path.join(samplesDir, file));
+    if (stats.isDirectory()) {
+      const matchResult = file.match(/^v([0-9]*)$/);
+      if (matchResult !== null) {
+        dirs.push(parseInt(matchResult[1]));
+      }
+    }
+  }
+  if (dirs.length === 0) {
+    return `${rootDir}/samples`;
+  } else {
+    return `${rootDir}/samples/v${Math.max(...dirs)}`;
```

Comment:
This is an interesting approach, but I think you can rely on sorting the directories lexically. `dirs.sort().slice(-1).pop()`